smithy-model/src/main/java/software/amazon/smithy/model/transform/ModelTransformer.java

```diff
@@ -481,6 +481,18 @@ public Model changeShapeType(Model model, Map<ShapeId, ShapeType> shapeToType) {
         return new ChangeShapeType(shapeToType).transform(this, model);
     }
 
+    /**
+     * Changes each compatible string shape with the enum trait to an enum shape.
+     *
+     * <p>Strings with enum traits that don't define names are not converted.
```

Comment:
Not safely, at least not in all cases. Those values could be literally anything that fits in a string. Automatically converting them would be risky because we could introduce collisions. We could convert those that are safely convertible though.

Reply:
I think we should have something that does a best effort through at least an opt-in. Otherwise no AWS SDK team could use this I think.